src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -64,11 +73,21 @@ withMiningCoordination logger enabled cutDb inner
     | not enabled = inner Nothing
     | otherwise = do
         t <- newTVarIO mempty
-        inner . Just $ MiningCoordination
+        fmap snd . concurrently (prune t) $ inner . Just $ MiningCoordination
             { _coordLogger = logger
             , _coordCutDb = cutDb
-            , _coordState = t
-            }
+            , _coordState = t }
+  where
+    prune :: TVar MiningState -> IO a
+    prune t = do
+        let !d = 600000000  -- 10 minutes
+        threadDelay d
+        ago <- over (_Unwrapped . _Unwrapped) (subtract (int d)) <$> getCurrentTimeIntegral
```

Comment:
Are you aware of Using functions from https://github.com/kadena-io/chainweb-node/blob/master/src/Numeric/AffineSpace.hs? Using it may make arithmetic over time values more readable. But it's at matter of style and preference.

Reply:
We might revisit this later.